### PR TITLE
Release v0.17.0

### PR DESCRIPTION
**Pull request by thanethomson**
📖 [CHANGELOG](https://github.com/informalsystems/tendermint-rs/blob/release/v0.17.0/CHANGELOG.md#v0170)

📖 [New "Releases" section in contributing guidelines](https://github.com/informalsystems/tendermint-rs/blob/release/v0.17.0/CONTRIBUTING.md#releases)

* [ ] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Updated CHANGELOG.md

### CODE REVIEW

**Comment by tony-iqlusion** [comment]
Is it still possible to get one last minute change in?

I'd like to cut another breaking release of the `k256` crate later today and bump it again, similar to https://github.com/informalsystems/tendermint-rs/pull/724

**Review by greg-szabo — DISMISSED** [review]
As I mentioned in our discussion, I would rather have a 0.16.0 -> 0.17.0 full changelog on top of the rc release changelogs (we should keep those too), just so we see all the changes going into the new release but it's just a personal preference.

Thanks for the doctest improvements and the contribution file additions!

**Review by ebuchman — DISMISSED** [review]
Awesome work.

**Comment by tony-iqlusion** [comment]
I opened a PR to bump the `k256` crate to the latest v0.7 release: https://github.com/informalsystems/tendermint-rs/pull/752

If it's possible to get that PR into this release, that'd be great!

File: CHANGELOG.md [hunk 1]
```diff
@@ -144,19 +37,24 @@ testing and greater compatibility with dependent projects.
...
   channels. ([#516])
-  
-- `[tendermint]` Removed all trace of Amino, including `amino_types` modules. 
-   All types are now `DomainType`s implementing Protobuf-encoding using Prost. 
-   ([#504], [#535], [#536], [#585])
-
-- `[tendermint]` Protocol breaking changes for compatibility with Tendermint Core v0.34 (and the Cosmos Stargate release) ([#305]):
-    - Validators are now sorted by voting power (descending) 
+- `[tendermint]` Removed all traces of Amino, including `amino_types` modules.
+  All types are now `DomainType`s implementing Protobuf-encoding using Prost.
```
Comment (greg-szabo) [inline CHANGELOG.md:41]:
```suggestion
  All types are now "domain types" implementing the `Protobuf` trait for Protobuf-encoding using Prost.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -144,19 +37,24 @@ testing and greater compatibility with dependent projects.
...
+  - Fix hash of empty Merkle tree to comply with RFC6962 ([#498])
+  - All binary encoding is done via protobuf3 instead of amino
+    ([#504], [#535], [#536], [#585])
+  - Various updates to JSON encoding ([#505])
+- `[tendermint]` Direct serialization capabilities have been removed from the
+  domain types. ([#639])
+- `[tendermint]` Work has started on making it compulsory to construct domain
+  types by way of their constructors to ensure validity. ([#639])
+- `[tendermint-proto]` The `DomainType` trait has now been renamed to
+  `Protobuf` to clarify its purpose throughout the codebase. ([#672])
```
Comment (greg-szabo) [inline CHANGELOG.md:57]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -165,60 +63,86 @@ testing and greater compatibility with dependent projects.
...
+  WebSocket endpoint). This aims to help improve testing and compatibility
+  between Tendermint in Go and Rust. ([#653])
+- `[rpc]` The `WebSocketClient` now adds support for all remaining RPC requests
+  by way of implementing the `Client` trait ([#646])
+- `[rpc]` Support for the `tx_search` RPC endpoint has been added ([#701])
+- `[rpc]` Responses that include events now automatically have their tag
+  key/value pairs decoded from base64, where previously tag key/value pairs
+  were Base64-encoded ([#717])
+- `[rpc]` Support for the `consensus_state` RPC endpoint has been added ([#719])
+- `[tendermint]` Bech32 encoding fix ([#690])
```
Comment (greg-szabo) [inline CHANGELOG.md:102]:
```suggestion
```
(This bug+fix was within the rc releases. Between 0.16->0.17, this didn't actually change.)

File: CHANGELOG.md [hunk 4]
```diff
@@ -165,60 +63,86 @@ testing and greater compatibility with dependent projects.
...
 ### BUG FIXES:
 
+- `[light-client]` Fix bug where a commit with only absent signatures would be
+  deemed valid instead of invalid ([#650])
+- `[light-client]` Revert a change introduced in [#652] that would enable DoS attacks,
+  where full nodes could spam the light client with massive commits (eg. 10k validators).
 - `[rpc]` Correctly handles control and keep-alive messages ([#516], [#590])
 - `[rpc]` More robust handling of concurrency issues ([#311], [#313])
+- `[tendermint]` CanonicalBlockId is now correctly decoded to `None`
+  in `CanonicalVote` and `CanonicalProposal` when its hash is empty.
```
Comment (greg-szabo) [inline CHANGELOG.md:123]:
```suggestion
```
I have a feeling this wasn't a problem in 0.16 either.

File: CHANGELOG.md [hunk 5]
```diff
@@ -165,60 +63,86 @@ testing and greater compatibility with dependent projects.
...
+- `[tendermint]` Time serialization fix (part of [#665])
+- `[tendermint]` Fix abci::Data serialization to base64-encoded string. ([#667])
+- `[tendermint]` Simplify abci::Log serialization ([#667])
+- `[tendermint]` consensus::State backwards compatibility for deserialization ([#679])
+- `[tendermint-proto]` Timestamp serialization fix (part of [#665])
```
Comment (greg-szabo) [inline CHANGELOG.md:128]: I guess these might have also been opened because of the upgrade, although there might be fixes in there. I'm not sure.

**Review by greg-szabo — APPROVED** [review]
Cool stuff. Made some suggestions to make it a bit more concise but they're not that important.

File: CHANGELOG.md (hunk 2 above)
Comment (thanethomson) [reply CHANGELOG.md:57]: Good catch! I wonder how this happened...

File: CHANGELOG.md (hunk 5 above)
Comment (thanethomson) [reply CHANGELOG.md:123]: We could easily take them out if you want?

```suggestion
```

**Review by greg-szabo — DISMISSED** [review]

**Review by greg-szabo — APPROVED** [review]
Script looks good. I like that it has a timeout.

**Pull request merged**